src/common/Foldable.ml

```diff
@@ -0,0 +1,119 @@
+(* This module follows the Jane Street organisation and naming conventions for 
+'standard interfaces' i.e. commonly used signatures; briefly:
+
+- `Basic` defines the _minimal_ set of type and function signatures that a
+user must provide
+- `S` defines the _complete_ set of type and functions signatures that will be
+exposed by any module supporting the  `standard interface` associated with `S`
+- `Make` is a functor (i.e. a function from module to module) that takes the 
+user supplied `Basic` module and returns a completed module with the signature
+defined by `S`
+
+Where a standard interface is defined for difference arities of 
+type constructors, the convention is to append the arity to the names of the 
+signatures and functors e.g. `Basic2`, `S2` and `Make2`
```

Comment:
This is awesome doc! thank you :) IF there are links to related Jane Street doc that would be good to include too.

Reply:
Not that I know of; I spent ages reading `Core` code and asked a couple of questions on OCaml discuss to work out what was going on initially!